cl/search/tests/tests_es_recap.py

```diff
@@ -3090,10 +3176,153 @@ def test_date_filed_sorting_function_score(self) -> None:
                 self._test_results_ordering(test, "docket_id")
 
         with self.captureOnCommitCallbacks(execute=True):
-            docket_recent.delete()
+            docket_entry_recent.docket.delete()
             docket_old.delete()
             docket_null_date_filed.delete()
 
+    @override_settings(SEARCH_API_PAGE_SIZE=1)
+    @mock.patch(
+        "cl.search.api_views.DocketESResultSerializer",
+        new=DocketESResultSerializerTest,
+    )
+    @mock.patch(
+        "cl.search.api_views.BaseRECAPDocumentESResultSerializer",
+        new=BaseRECAPDocumentESResultSerializerTest,
+    )
+    @mock.patch(
+        "cl.search.api_views.RECAPESResultSerializer",
+        new=RECAPESResultSerializerTest,
+    )
+    @mock.patch(
+        "cl.search.api_utils.set_results_highlights",
+        side_effect=mock_set_results_highlights,
+    )
+    def test_stable_scores_date_sort(
+        self, mock_set_results_highlights
+    ) -> None:
+        """Test if the function's score remains stable when used for sorting by
+        dateFiled asc and entry_date_filed asc across pagination, avoiding
+        the impact of time running on the scores.
+        """
+
+        # Query string, order by dateFiled desc
+        search_params = {
+            "q": "SUBPOENAS SERVED",
+            "order_by": "dateFiled asc",
+            "highlight": False,
+        }
+
+        # string query date_filed asc
+        date_filed_asc_string_query = search_params
+        # match_all query date_filed asc
+        date_filed_asc_match_all = search_params.copy()
+        del date_filed_asc_match_all["q"]
+        # string query entry_date_filed asc
+        entry_filed_asc_string_query = search_params.copy()
+        entry_filed_asc_string_query["order_by"] = "entry_date_filed asc"
+        # match_all query entry_date_filed asc
+        entry_filed_asc_match_all = entry_filed_asc_string_query.copy()
+        del entry_filed_asc_match_all["q"]
+
+        tests_params = [
+            date_filed_asc_string_query,
+            date_filed_asc_match_all,
+            entry_filed_asc_string_query,
+            entry_filed_asc_match_all,
+        ]
+        # Generate tests for all the search types.
+        all_tests = []
+        for test_param in tests_params:
+            for search_type in [
+                SEARCH_TYPES.RECAP,
+                SEARCH_TYPES.DOCKETS,
+                SEARCH_TYPES.RECAP_DOCUMENT,
+            ]:
+                test_param_copy = test_param.copy()
+                test_param_copy["type"] = search_type
+                all_tests.append(test_param_copy)
+
+        for search_params in all_tests:
+            with self.subTest(
+                search_params=search_params, msg="Test stable scores."
+            ):
+                original_datetime = now().replace(day=1, hour=5, minute=0)
```

Comment:
I think we can get this variable out of the loop, right? We don't need to recompute it for each subtest.

Reply:
Yeah, thanks for noticing it. I have moved the variable out of the loop.